controllers/admin/AdminCartRulesController.php

```diff
@@ -60,6 +60,95 @@ public function __construct()
         ];
     }
 
+    /**
+     * Function used to render the list to display for this controller.
+     * Add joins to show only cart_rules to which the employee has access, restrict edit and delete action
+     *
+     * @return string|false
+     *
+     * @throws PrestaShopException
+     */
+    public function renderList()
+    {
+        if (!($this->fields_list && is_array($this->fields_list))) {
+            return false;
+        }
+
+        $context = Context::getContext();
+        $all_shops = Shop::getCompleteListOfShopsID();
+        if ($context->employee->isSuperAdmin()) {
+            $employee_shops = $all_shops;
+        } else {
+            $employee_shops = $context->employee->getAssociatedShops();
+        }
+        $context_shops = Shop::getContextListShopID();
+        $employee_has_full_access = empty(array_diff($all_shops, $employee_shops));
```

Comment:
What would return `$context->employee->getAssociatedShops()` on a SuperAdmin employee ?

 Cause if I understand well what you are trying to do, I really think something like :
 ```php
$all_shops = Shop::getCompleteListOfShopsID();
$employee_shops = $context->employee->getAssociatedShops();
$context_shops = Shop::getContextListShopID();
$employee_has_full_access = (count($all_shops) == count($employee_shops));
 ```
 would be clearer and faster.
 
But my intuition is that `$context->employee->getAssociatedShops()` for SuperAdmin employee isn't right.
In the same way, `$employee_has_full_access` is an information from employee (sounds better to be able to call  `$context->employee->hasShopFullAccess()`).

It is not normal that you have to recreate this information dynamically. It would make understanding heavier and induce bugs.

What do you think @PrestaShop/prestashop-core-developers  ?
When I searched a bit in the code, I saw lots of those "tricks". Maybe it is a bit out of this PR scope but as it is for develop branch, I really prefer to review `EmployeeCore` object ^^'
Furthermore, I think it could be done without adding BC, maybe some deprecation :thinking: 

Maybe some refactoring is already planned / done ?

Reply:
>  But my intuition is that $context->employee->getAssociatedShops() for SuperAdmin employee isn't right.

Yes, that's why 
```
        if ($context->employee->isSuperAdmin()) {
            $employee_shops = $all_shops;
        } else {
            $employee_shops = $context->employee->getAssociatedShops();
        }
```

> Cause if I understand well what you are trying to do, I really think something like :
```
$all_shops = Shop::getCompleteListOfShopsID();
$employee_shops = $context->employee->getAssociatedShops();
$context_shops = Shop::getContextListShopID();
$employee_has_full_access = (count($all_shops) == count($employee_shops));
```
>would be clearer and faster.

This would be wrong for my use, because an Administrator has access to all shops, even though `$context->employee->getAssociatedShops();` might not return all shop in all cases (but most of the time it does).

> `$context->employee->hasShopFullAccess()`

Yes would be nice that existed. But do you think this bit of code is required throughout the code? I don't think so. I think I'm doing something special here.

> Maybe it is a bit out of this PR scope but as it is for develop branch, I really prefer to review EmployeeCore object ^^'

That would be great, if you would do this.